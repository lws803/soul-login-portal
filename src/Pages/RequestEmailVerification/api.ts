import axios from 'axios';
import axiosRetry from 'axios-retry';

import { BASE_URL } from 'src/constants';
import { ApiError } from 'src/shared/apiTypes';

axiosRetry(axios, { retryDelay: axiosRetry.exponentialDelay, retries: 3 });

export const sendEmailConfirmation = async ({ email }: { email: string }) => {
  try {
    await axios.post(
      `${BASE_URL}/v1/users/resend-confirmation-token`,
      {},
      { params: { email } },
    );
    return { error: null };
  } catch ({ response: { data } }) {
    return { error: data as ApiError };
  }
};
