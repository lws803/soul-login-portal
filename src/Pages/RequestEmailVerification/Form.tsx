import { useFormik } from 'formik';
import * as Yup from 'yup';
import {
  FormControl,
  FormLabel,
  Input,
  FormErrorMessage,
  HStack,
  Button,
} from '@chakra-ui/react';

import { sendEmailConfirmation } from './api';

export default function Form({ setErrors, setIsSuccess }: Props) {
  const formik = useFormik({
    initialValues: {
      email: '',
    },
    onSubmit: async (values) => {
      setErrors([]);
      const { error } = await sendEmailConfirmation(values);

      if (error) {
        setErrors([error.message]);
      } else {
        setIsSuccess(true);
      }
    },
    validationSchema: Yup.object({
      email: Yup.string().email().required(),
    }),
    validateOnChange: false,
  });

  return (
    <form onSubmit={(e) => formik.handleSubmit(e as any)}>
      <FormControl
        isInvalid={!!formik.errors.email && formik.touched.email}
        marginTop={8}
      >
        <FormLabel htmlFor="email">Email</FormLabel>
        <Input
          id="email"
          name="email"
          onChange={formik.handleChange}
          value={formik.values.email}
          aria-label="Email input"
          variant="filled"
          placeholder="Enter your email"
        />
        {formik.errors.email && (
          <FormErrorMessage>{formik.errors.email}</FormErrorMessage>
        )}
      </FormControl>
      <HStack mt={8}>
        <Button
          type="submit"
          isLoading={formik.isSubmitting}
          loadingText="Sending verification email"
          width="100%"
          bg="soul.pink.light"
          _hover={{ bg: 'soul.pink.dark' }}
          _active={{ bg: 'soul.pink.dark' }}
        >
          Request email verification
        </Button>
      </HStack>
    </form>
  );
}

type Props = {
  setErrors: (errors: string[]) => void;
  setIsSuccess(isSuccess: boolean): void;
};
