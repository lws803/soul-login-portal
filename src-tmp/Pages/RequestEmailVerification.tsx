import React, { useState } from 'react';

import Form from './RequestEmailVerification/Form';

import Status from '../components/Status';
import Page from '../components/Page';
import Title from '../components/Title';

export default function RequestEmailVerification() {
  const [errors, setErrors] = useState<string[]>([]);
  const [isSuccess, setIsSuccess] = useState(false);

  return (
    <Page errors={errors}>
      {isSuccess ? (
        <Status
          status="Request submitted!"
          message="Sent email verification, please check your email."
        />
      ) : (
        <>
          <Title title="Resend email verification" />
          <Form setErrors={setErrors} setIsSuccess={setIsSuccess} />
        </>
      )}
    </Page>
  );
}
