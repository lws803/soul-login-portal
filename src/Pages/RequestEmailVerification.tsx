import React, { useState } from 'react';

import Form from './RequestEmailVerification/Form';

import Status from '../components/Status';
import Page from '../components/Page';
import Title from '../components/Title';

export default function RequestEmailVerification() {
  const [errors, setErrors] = useState<string[]>([]);
  const [isSuccess, setIsSuccess] = useState(false);

  return (
    <Page errors={errors} title={<Title title="Resend email verification" />}>
      {isSuccess ? (
        <Status
          status="Request submitted!"
          message={
            'Email verification requested, you will receive an email shortly if your ' +
            'email address exists in our database. Please check your spam inbox if ' +
            "you haven't received an email after a few minutes."
          }
        />
      ) : (
        <Form setErrors={setErrors} setIsSuccess={setIsSuccess} />
      )}
    </Page>
  );
}
