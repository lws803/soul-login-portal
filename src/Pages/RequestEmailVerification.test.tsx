import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import RequestEmailVerification from './RequestEmailVerification';

describe(RequestEmailVerification, () => {
  it('renders', async () => {
    const { container } = render(
      <MemoryRouter>
        <RequestEmailVerification />
      </MemoryRouter>,
    );

    expect(container).toMatchSnapshot();
  });
});
