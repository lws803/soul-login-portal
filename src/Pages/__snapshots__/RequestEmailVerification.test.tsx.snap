// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestEmailVerification renders 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      class="css-1wiwcvy"
    >
      <div
        class="css-1ryq4o3"
      >
        <div
          class="css-1yjvs5a"
        >
          <p
            class="chakra-text css-odt1ir"
          >
            Resend email verification
          </p>
          <div
            class="css-164r41r"
          />
        </div>
        <form>
          <div
            class="chakra-form-control css-164r41r"
            role="group"
          >
            <label
              class="chakra-form__label css-uvho5g"
              for="email"
              id="field-1-label"
            >
              Email
            </label>
            <input
              aria-label="Email input"
              class="chakra-input css-0"
              id="email"
              name="email"
              placeholder="Enter your email"
              value=""
            />
          </div>
          <div
            class="chakra-stack css-nzla2i"
          >
            <button
              class="chakra-button css-a0tdue"
              type="submit"
            >
              Request email verification
            </button>
          </div>
        </form>
      </div>
    </div>
    <div
      class="css-l5skqa"
    >
      <div
        class="chakra-stack css-12ogrgp"
      >
        <a
          class="chakra-link css-epvm6"
          href="https://github.com/soul-project/login-portal"
          target="_blank"
        >
          Github
        </a>
        <a
          class="chakra-link css-epvm6"
          href="https://github.com/soul-project/soul"
          target="_blank"
        >
          Soul Network's Github
        </a>
        <a
          class="chakra-link css-epvm6"
          href="mailto:lws803@gmail.com"
          target="_blank"
        >
          Contact me
        </a>
      </div>
    </div>
  </div>
</div>
`;
